### PR TITLE
Use version as build qualifier for stable/release builds

### PR DESCRIPTION
**Pull request by kthoms**
For stable/release builds take the version as qualifier. Nightly can
stay as is.
- fixes bug#522525
- added method getBuildQualifier
- fix typo in method name generatePromoteProperties
- upgrade gradle wrapper to 5.4.1

See https://www.eclipse.org/modeling/tmf/downloads/ for 2.18.0.M3 deployment

### CODE REVIEW

File: buildSrc/.classpath [hunk 1]
```diff
@@ -3,7 +3,12 @@
 	<classpathentry kind="src" path="src/main/java"/>
 	<classpathentry kind="src" path="build/xtend/main"/>
 	<classpathentry kind="src" path="src/main/resources"/>
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/">
+		<attributes>
+			<attribute name="module" value="true"/>
```
Comment (cdietrich) [inline buildSrc/.classpath]: ?

Comment (kthoms) [reply buildSrc/.classpath]: some automatic added stuff. I'll remove it from this change.

File: buildSrc/build.gradle [hunk 2]
```diff
@@ -5,7 +5,7 @@ buildscript {
 	}
 }
 
-ext.xtextVersion = '2.10.0'
+ext.xtextVersion = '2.17.0'
```
Comment (cdietrich) [inline buildSrc/build.gradle:8]: bump the xtend plugin too. version is 2.0.4
maybe bump gradle version (wrapper) as well

Comment (kthoms) [reply buildSrc/build.gradle:8]: gradle wrapper is bumped, or do you mean sthg else?

Comment (cdietrich) [reply buildSrc/build.gradle:8]: did not see the wrapper. but the plugin is the old

Comment (kthoms) [reply buildSrc/build.gradle:8]: already changed, see new commit

**Review by cdietrich — APPROVED** [review]

File: buildSrc/build.gradle (hunk 2 above)
Comment (cdietrich) [reply buildSrc/build.gradle:8]: did you run the wrapper command twice?

Comment (cdietrich) [reply buildSrc/build.gradle:8]: (would have expected changes to sh and bat file too

**Pull request merged**